src/number_theory/number_field.lean

```diff
@@ -249,4 +250,83 @@ end
 
 end roots
 
+section bounded
+
+open finite_dimensional polynomial set
+
+variables {K : Type*} [field K] [number_field K]
+variables {A : Type*} [normed_field A] [is_alg_closed A] [normed_algebra ℚ A]
+
+lemma coeff_bdd_of_norm_le {B : ℝ} {x : K} (h : ∀ φ : K →+* A, ∥φ x∥ ≤ B) (i : ℕ):
+  ∥(minpoly ℚ x).coeff i∥ ≤ (max B 1) ^ (finrank ℚ K) * (finrank ℚ K).choose ((finrank ℚ K) / 2) :=
+begin
+  have hx : is_integral ℚ x := is_separable.is_integral _ _,
+  rw (by rw [coeff_map, norm_algebra_map'] :
+    ∥(minpoly ℚ x).coeff i∥ = ∥(map (algebra_map ℚ A) (minpoly ℚ x)).coeff i∥),
+  refine coeff_bdd_of_roots_le _ (minpoly.monic hx) (is_alg_closed.splits_codomain _)
+    (minpoly.nat_degree_le hx) _ i,
+  intros z hz,
+  rsuffices ⟨φ, rfl⟩ : ∃ (φ : K →+* A), φ x = z, {exact h φ },
+  letI : char_zero A := char_zero_of_injective_algebra_map (algebra_map ℚ _).injective,
+  rw [← set.mem_range, rat_range_eq_roots, mem_root_set_iff, aeval_def],
+  convert (mem_roots_map _).mp hz,
+  repeat { exact monic.ne_zero (minpoly.monic hx), },
+end
+
+end bounded
+
+section finite
+
+open finite_dimensional polynomial set
+
+variables (K : Type*) [field K] [number_field K]
+variables (A : Type*) [normed_field A] [is_alg_closed A] [normed_algebra ℚ A]
```

Comment:
You just declared these in the previous section, you can avoid ding this twice just moving some lines (you can create a titleless section with `section` if you don't want the variables to be defined forever in the file). To make them explicit you can use `variables (K A).`

Reply:
I suspected there was a way to do that. Thanks!